### PR TITLE
WIP: [LEVWEB-792] Use Oracle as the source database

### PR DESCRIPTION
**Pull request by daniel-ac-martin**
This is probably a more typical use-case and allows us to side-step what
appears to be a bug in the MySQL support.

The downside is that the Oracle image is ugly and listens on it's port
before it is actually ready to accept connections, requiring us to sleep
for a certain time to allow for this.

### CODE REVIEW

File: oracle-source/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,3 @@
+FROM sath89/oracle-12c:latest
```
Comment (easternbloc) [inline oracle-source/Dockerfile:1]: 12c? what about 11g?

File: entrypoint.sh [hunk 2]
```diff
@@ -46,10 +46,11 @@ case "${DB_TYPE}" in
     JDBC_URL="jdbc:mysql://${DB_HOST}:${DB_PORT}/${DB_NAME}"
     ;;
   "oracle")
-    DB_PORT="${DB_PORT:-1575}"
-    DB_CMD="sqlplus64 \"${DB_USER}/${DB_PASS}@//${DB_HOST}/${DB_NAME}\""
+    DB_PORT="${DB_PORT:-1521}"
+    DB_CMD="sqlplus64 \"${DB_USER}/${DB_PASS}@//${DB_HOST}:${DB_PORT}/${DB_NAME}\""
     JDBC_DRIVER="oracle.jdbc.driver.OracleDriver"
     JDBC_URL="jdbc:oracle:thin:@${DB_HOST}:${DB_PORT}:${DB_NAME}"
+    sleep 480 # Unfortunately the Oracle image listens on its port before it is ready to accept connections
```
Comment (easternbloc) [inline entrypoint.sh:53]: haha! Amazing :)

**Review by easternbloc — APPROVED** [review]
Really happy we've got this ironed out.

Maybe in the future we can add another `source` that's a postgres like we had running yesterday.

Either way I approve of these changes other than the question around using 12c?

**Comment by daniel-ac-martin** [comment]
This still needs testing which is why it's just a WIP. - I ran out of time last night.

I'm not sure it's a good idea because of the need to sleep.

**Comment by daniel-ac-martin** [comment]
I'm not to fussed about what the source database as it is SymmetricDS responsibility to support them. We really just need to prove the concept.

Arguably, we should have multiple docker-compose to demonstrate each case but it increases the maintenance burden.

**Comment by daniel-ac-martin** [comment]
Another option would be to use a 'nice' source database in the docker-compose and a more typical database in the k8s configuration we intend to add.

**Comment by easternbloc** [comment]
I'm gonna close this now because it's great and all but it's not in line with how we want to present the example setup.

**Pull request closed without merging**